src/tiled/automappingmanager.cpp

```diff
@@ -204,22 +207,46 @@ bool AutomappingManager::loadFile(const QString &filePath)
 
 void AutomappingManager::setMapDocument(MapDocument *mapDocument)
 {
-    cleanUp();
-    if (mMapDocument)
+    QString oldRules;
+    if (mMapDocument) {
+        oldRules = rulesFileName();
         mMapDocument->disconnect(this);
+    }
 
     mMapDocument = mapDocument;
 
+    QString newRules;
     if (mMapDocument) {
-        connect(mMapDocument, &MapDocument::regionEdited,
-                this, &AutomappingManager::onRegionEdited);
+        newRules = rulesFileName();
+        if (newRules != oldRules) {
+            cleanUp();
+            if (!mWatcher.files().isEmpty())
+                mWatcher.removePaths(mWatcher.files());
+        }
+        if (QFile::exists(newRules))
+            mWatcher.addPath(newRules);
```

Comment:
It doesn't really make sense to watch the file if it hasn't been loaded yet, so I would move this to the `loadFile` function. 

In addition, that will fix the reload in case the user modifies any rules file that is referenced by the main rules file (it is possible to include another rules file in a rules file by just listing a `.txt` file).

Finally, I guess it would make sense to move the clearing of `mWatcher` to the `cleanUp` function. Once the loaded rules have been cleaned, it is no longer interesting to watch any files for changes after all.